modAL/models/base.py

```diff
@@ -389,51 +278,51 @@ def query(self, X_pool, *query_args, **query_kwargs) -> Union[Tuple, modALinput]
             Return value of the query_strategy function. Should be the indices of the instances from the pool chosen to
             be labelled and the instances themselves. Can be different in other cases, for instance only the instance to
             be labelled upon query synthesis.
+            query_metrics: returns also the corresponding metrics, if return_metrics == True
         """
-        query_result = self.query_strategy(self, X_pool, *query_args, **query_kwargs)
 
-        if isinstance(query_result, tuple):
-            warnings.warn("Query strategies should no longer return the selected instances, "
-                          "this is now handled by the query method. "
-                          "Please return only the indices of the selected instances", DeprecationWarning)
-            return query_result
-
-        return query_result, retrieve_rows(X_pool, query_result)
-
-    def rebag(self, **fit_kwargs) -> None:
-        """
-        Refits every learner with a dataset bootstrapped from its training instances. Contrary to .bag(), it bootstraps
-        the training data for each learner based on its own examples.
+        try:
+            query_result, query_metrics = self.query_strategy(
+                self, X_pool, *query_args, **query_kwargs)
 
-        Todo:
-            Where is .bag()?
+        except ValueError:
+            query_metrics = None
+            query_result = self.query_strategy(
+                self, X_pool, *query_args, **query_kwargs)
```

Comment:
Same as above for the `BaseLearner` Do we always get `ValueError` here if something goes wrong?.